### ISSUE TITLE
Possible Null Reference on `t?.ToString()` for unconstrained T

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#28792]
```C#
class C
{
	void M<T>(T t) => t?.ToString();
}
```

Expected: There should be no warning
Actual: `warning CS8602: Possible dereference of a null reference.`

FYI @jcouv @cston

**Comment by jcouv** [comment]
Closing as fixed by #28722

**Comment by jcouv** [comment]
This was fixed in Aleksey's latest change (https://github.com/dotnet/roslyn/pull/30913). We'll just need to verify there's a test (or add one).
```C#

        [Fact]
        public void TEST()
        {
            var source0 =
@"
class C
{
	void M<T>(T t) => t?.ToString();
}";
            var comp0 = CreateCompilation(new[] { source0 }, options: WithNonNullTypesTrue());
            comp0.VerifyDiagnostics(
                );
        }
```
